Remove unmaintained nujson from benchmark and bump to Python 3.11

hugovk opened the pull request:
The benchmark CI job has been failing for a couple of months:

https://github.com/ultrajson/ultrajson/actions/workflows/benchmark.yml

Because it's failing to build the wheel for nujson:

<details>
<summary>Details</summary>

```
Building wheels for collected packages: nujson
  Building wheel for nujson (pyproject.toml): started
  Building wheel for nujson (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Building wheel for nujson (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [50 lines of output]
      Warning: 'classifiers' should be a list, got type 'filter'
      running bdist_wheel
      running build
      running build_ext
      Traceback (most recent call last):
Failed to build nujson
        File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 416, in build_wheel
          return self._build_with_temp_dir(['bdist_wheel'], '.whl',
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 401, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super(_BuildMetaLegacyBackend,
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 77, in <module>
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 1244, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 3[43](https://github.com/ultrajson/ultrajson/actions/runs/5141046608/jobs/9253110620#step:4:44), in run
          self.run_command("build")
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 12[44](https://github.com/ultrajson/ultrajson/actions/runs/5141046608/jobs/9253110620#step:4:45), in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 1244, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-fpxckxlz/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line [68](https://github.com/ultrajson/ultrajson/actions/runs/5141046608/jobs/9253110620#step:4:69), in run
      ModuleNotFoundError: No module named 'numpy'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for nujson
ERROR: Could not build wheels for nujson, which is required to install pyproject.toml-based projects
```

</details>

However, https://pypi.org/project/nujson/ hasn't had any releases since December 2019 and https://github.com/caiyunapp/ultrajson has been archived since August 2022.

Let's just remove it.

I updated the benchmark results on the README from a CI run: https://github.com/hugovk/ultrajson/actions/runs/5145083552/jobs/9262222063

hugovk committed:
Remove unmaintained nujson from benchmark and bump to Python 3.11

hugovk committed:
Add build trigger

hugovk committed:
Update benchmark results from CI run

re: https://github.com/hugovk/ultrajson/actions/runs/5145083552/jobs/9262222063

bwoodsend commented on README.md:
```diff
@@ -85,44 +85,46 @@ specified below each.
-| encode                                                                        |        533 |        582 |            |        408 |        431 |
-| decode                                                                        |        466 |        454 |            |        154 |        164 |
+- CPython 3.11.3 (main, Apr  6 2023, 07:55:46) [GCC 11.3.0]
+- ujson        : 0.1.dev1
```
Looks like setuptools-scm is having a bad day.

hugovk replied:
Ah yes, missing the Git tags. Updated!

hugovk committed:
Fetch all tags for setuptools_scm

hugovk committed:
Update benchmark results from CI run

re: https://github.com/hugovk/ultrajson/actions/runs/5148167409/jobs/9269512936

JustAnotherArchivist commented on README.md:
```diff
@@ -85,44 +85,46 @@ specified below each.
-| encode                                                                        |        533 |        582 |            |        408 |        431 |
-| decode                                                                        |        466 |        454 |            |        154 |        164 |
+- CPython 3.11.3 (main, Apr  6 2023, 07:55:46) [GCC 11.3.0]
+- ujson        : 1.36.dev558
```
Seems like it's using the v1.35 tag and ignoring the higher version tags without the `v` prefix.

hugovk replied:
Aha, it's because I ran it on my branch and the tags weren't in sync with upstream.

I'll redo it, and also use `run: git fetch --prune --unshallow`, which is what we have for `deploy.yml`

hugovk replied:
https://github.com/hugovk/ultrajson/actions/runs/5148581127/jobs/9270556867 running with `ujson-5.7.1.dev26`, will update the benchmark stats here tomorrow 👍

hugovk replied:
Updated!

hugovk committed:
'git fetch --prune --unshallow' like deploy.yml

hugovk committed:
Update benchmark results from CI run

re: https://github.com/hugovk/ultrajson/actions/runs/5148581127/jobs/9270556867
